frontend/components/adopter/ViewPets/data/filters.ts

```diff
@@ -0,0 +1,196 @@
+import { Animal } from "@contract";
+import dayjs from "dayjs";
+import customParseFormat from "dayjs/plugin/customParseFormat";
+import utc from "dayjs/plugin/utc";
+
+dayjs.extend(customParseFormat);
+dayjs.extend(utc);
+
+export const SpeciesFilterOptions = [
+    {
+        options: [
+            {
+                label: "Cat",
+                value: Animal.Species.Cat,
+            },
+            {
+                label: "Dog",
+                value: Animal.Species.Dog,
+            },
+            {
+                label: "Others",
+                value: Animal.Species.Others,
+            },
+        ],
+    },
+];
+
+export enum AgeFilter {
+    Above0Months,
+    Above6Months,
+    Above12Months,
+    Above36Months,
+    Above72Months,
+    Above120Months,
+}
+
+export const AgeFilterOptions = [
+    {
+        title: "Under a year",
+        options: [
+            {
+                label: "1-6 months",
+                value: AgeFilter.Above0Months,
+            },
+            {
+                label: "7-12 months",
+                value: AgeFilter.Above6Months,
+            },
+        ],
+    },
+    {
+        title: "Beyond a year",
+        options: [
+            {
+                label: "1-3 years",
+                value: AgeFilter.Above12Months,
+            },
+            {
+                label: "4-6 years",
+                value: AgeFilter.Above36Months,
+            },
+            {
+                label: "6-9 years",
+                value: AgeFilter.Above72Months,
+            },
+            {
+                label: "10 years and above",
+                value: AgeFilter.Above120Months,
+            },
+        ],
+    },
+];
+
+export const AgeFilterMatcher = new Map<
```

Comment:
I played around with this and noticed this method does not enforce all the AgeFilter values to be a key. An alternative would be something like this:

```
export const _AgeFilterMatcher: Record<AgeFilter, (ageMonths: number) => boolean> ={
    [AgeFilter.Above0Months]: (ageMonths) => ageMonths >= 0 && ageMonths <= 6,
    [AgeFilter.Above6Months]: (ageMonths) => ageMonths >= 0 && ageMonths <= 6,
    [AgeFilter.Above12Months]: (ageMonths) => ageMonths >= 0 && ageMonths <= 6,
    [AgeFilter.Above36Months]: (ageMonths) => ageMonths >= 0 && ageMonths <= 6,
    [AgeFilter.Above72Months]: (ageMonths) => ageMonths >= 0 && ageMonths <= 6,
    [AgeFilter.Above120Months]: (ageMonths) => ageMonths >= 0 && ageMonths <= 6,
}
```